src/GNDStk/tree/test/assign.test.cpp

```diff
@@ -0,0 +1,105 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+
+
+
+// -----------------------------------------------------------------------------
+// assign(rhs,ossr)
+// -----------------------------------------------------------------------------
+
+// TREE = a existing "right-hand-side" tree that arrives here as something
+//        we'll use on the right-hand-side of an assignment to a new tree
+// M = metadata container class for the new tree
+// C = children container class for the new tree
+template<
+   class TREE,
+   template<class ...> class M = std::vector,
+   template<class ...> class C = std::vector
+>
+bool assign(
+   const TREE &rhs, // arriving tree
+   const std::ostringstream &ossr // the arriving tree printed to a string
+) {
+   // assignment: lhs = rhs
+   gnds::Tree<M,C> lhs;
+   lhs = rhs;
+
+   // print lhs to a string
+   std::ostringstream ossl;
+   ossl << lhs;
+
+   // compare the two strings
+   return ossl.str() == ossr.str();
+}
+
+
+
+// -----------------------------------------------------------------------------
+// assign()
+// -----------------------------------------------------------------------------
+
+// Tests various tree assignments, with a "right-hand-side" of Tree<M,C>
+// M = metadata container class
+// C = children container class
+template<
+   template<class ...> class M = std::vector,
+   template<class ...> class C = std::vector
+>
+bool assign()
+{
+   // read a meaningful GNDS hierarchy into the tree that
+   // we'll use as the right-hand-side of our assignments
+   const gnds::Tree<M,C> rhs("n-008_O_016.xml");
+
+   // print the tree to a string, which we'll use soon as
+   // something to which we'll compare other trees
+   std::ostringstream ossr;
+   ossr << rhs;
+
+   // assign the tree into other tree variants, and check
+   // that the contents of each such result are the same
+   return
```

Comment:
Did you write this remark before realizing that my template arguments specified container types, and not something else? These tests were essentially to see that I didn't inadvertently do something that would work one way in one container type than it did with another. Not that we'd really expect an issue, if the supported C++ container types are written consistently in the C++ library, but you never know what mistake could be made in something that's built on them. Imagine somebody who enters two pairs of values into a vector, then enters the same two pairs (a key/value) into a map. Oops, same key, and a probable code error is uncovered. I'm trying to cover as many bases as possible with these tests. Errors can be prone to happen in places for which somebody say, "oh, we don't have to worry about an error there!" :-)

Reply:
This was meant as a stress test, to ensure that ```Tree<[A[,B]]> = Tree<[C[,D]]>``` works for all combinations of the container types that are currently allowed for Tree. Writing out the ```13*13``` currently allowable cases would be reasonable; maybe not so much when I generalize Tree in order to allow more types of containers. The code is written in a way that basically uses combinatorics, so that I wrote ```13+13``` lines where there would otherwise have been ```13*13``` lines. It does look like this test, in its current form, may obfuscate the precise location of any errors, in terms of the way the Catch macros report things. I'll think about other ways to do this, so that any errors would be reported in a clearer fashion.